src/model_loader/state_dict.cpp

```diff
@@ -238,6 +238,20 @@ StateDict StateDict::select(const std::string_view& prefix) const {
   }
   return {std::move(selected), shard_id_, num_shards_};
 }
+// add the scalar value into the selected tensors ending with suffix
+StateDict StateDict::add_scalar_with_suffix(const std::string_view& suffix,
```

Comment:
we don't need to introduce a new function, let's use `select_with_transform` instead.